sapl/base/views.py

```diff
@@ -915,6 +919,339 @@ def get(self, request, *args, **kwargs):
+
+def legislatura_infindavel():
+    legislaturas = []
+
+    for legislatura in Legislatura.objects.all().order_by('-numero'):
+        if not legislatura.data_fim:
+            legislaturas.append(legislatura)
+
+    return legislaturas
+
+
+class ListarLegislaturaInfindavelView(PermissionRequiredMixin, ListView):
+    model = get_user_model()
+    template_name = 'base/legislatura_infindavel.html'
+    context_object_name = 'legislatura_infindavel'
+    permission_required = ('base.list_appconfig',)
+    paginate_by = 10
+
+    def get_queryset(self):
+        return legislatura_infindavel()
+
+    def get_context_data(self, **kwargs):
+        context = super(
+            ListarLegislaturaInfindavelView, self
+            ).get_context_data(**kwargs)
+        paginator = context['paginator']
+        page_obj = context['page_obj']
+        context['page_range'] = make_pagination(
+            page_obj.number, paginator.num_pages)
+        context[
+            'NO_ENTRIES_MSG'
+            ] = 'Nenhuma encontrada.'
+        return context
+
+
+def bancada_comissao_autor_externo():
+    lista_bancada_autor_externo = []
+    lista_comissao_autor_externo = []
+
+    tipo_autor_externo = TipoAutor.objects.filter(descricao='Externo')
+
+    for bancada in Bancada.objects.all().order_by('nome'):
+        autor_externo = bancada.autor.filter(tipo=tipo_autor_externo)
+
+        if len(autor_externo):
+            q_autor_externo = bancada.autor.get(tipo=tipo_autor_externo)
+            lista_bancada_autor_externo.append(
+                (q_autor_externo, bancada, 'Bancada', 'sistema/bancada')
+            )
+
+    for comissao in Comissao.objects.all().order_by('nome'):
+        autor_externo = comissao.autor.filter(tipo=tipo_autor_externo)
+
+        if len(autor_externo):
+            q_autor_externo = comissao.autor.get(tipo=tipo_autor_externo)
+            lista_comissao_autor_externo.append(
+                (q_autor_externo, comissao, 'Comissão', 'comissao')
+            )
+
+    return lista_bancada_autor_externo + lista_comissao_autor_externo
+
+
+class ListarBancadaComissaoAutorExternoView(PermissionRequiredMixin, ListView):
+    model = get_user_model()
+    template_name = 'base/bancada_comissao_autor_externo.html'
+    context_object_name = 'bancada_comissao_autor_externo'
+    permission_required = ('base.list_appconfig',)
+    paginate_by = 10
+
+    def get_queryset(self):
+        return bancada_comissao_autor_externo()
+
+    def get_context_data(self, **kwargs):
+        context = super(
+            ListarBancadaComissaoAutorExternoView, self
+            ).get_context_data(**kwargs)
+        paginator = context['paginator']
+        page_obj = context['page_obj']
+        context['page_range'] = make_pagination(
+            page_obj.number, paginator.num_pages)
+        context[
+            'NO_ENTRIES_MSG'
+            ] = 'Nenhum encontrado.'
+        return context
+
+
+def autores_duplicados():
+    lista_duplicados = []
```

Comment:
Todo código abaixo pode ser reescrito como:
```suggestion
def autores_duplicados():
     return [autor.values for autor in Autor.objects.values(
        'nome', 'tipo__descricao').annotate(
            count=Count('nome')).filter(count__gt=1)]
```

Reply:
Reescrito.